src/MongoDB.Driver/Encryption/LibMongoCryptControllerBase.cs

```diff
@@ -133,7 +135,33 @@ protected async Task<byte[]> ProcessStatesAsync(CryptContext context, string dat
         private IMongoCollection<BsonDocument> GetKeyVaultCollection()
         {
             var keyVaultDatabase = _keyVaultClient.GetDatabase(_keyVaultNamespace.DatabaseNamespace.DatabaseName);
-            return keyVaultDatabase.GetCollection<BsonDocument>(_keyVaultNamespace.CollectionName);
+
+            var collectionSettings = new MongoCollectionSettings
+            {
+                ReadConcern = ReadConcern.Majority,
+                WriteConcern = WriteConcern.WMajority
+            };
+            return keyVaultDatabase.GetCollection<BsonDocument>(_keyVaultNamespace.CollectionName, collectionSettings);
+        }
+
+        private void ParseEndpoint(string value, out string host, out int port)
```

Comment:
As an option, this method can be placed in `EndPointHelper`

Reply:
I am of the opinion that this method is a simple string manipulation problem and should simply be:

    private void ParseKmsEndpoint(string endpoint, out string host, out int port)
    {
        var match = Regex.Match(endpoint, @"^(?<host>.*):(?<port>\d+)$");
        if (match.Success)
        {
            host = match.Groups["host"].Value;
            port = int.Parse(match.Groups["port"].Value);
        }
        else
        {
            host = endpoint;
            port = 443;
        }
    }

The endpoint string is only of two forms: `host` or `host:port`. A simple test can determine which one it is and we can easily pick apart the `host` and the `port` parts.

There is no need to involve `EndPointHelper.Parse` in the interest of sharing code. `EndPointHelper.Parse` does *too much* work, and returns the wrong data type, which is evident because the returned `EndPoint` values don't have `Host` and `Port` properties, which we can only access through downcasting.

Dmitry does not agree with me on this (we have discussed it at length offline), so I would be interested in getting other reviewers' opinions on this.

Reply:
ok, I will change it.